keanu-project/src/test/java/io/improbable/keanu/tensor/bool/SimpleBooleanTensorTest.java

```diff
@@ -205,4 +206,55 @@ public void canReshape() {
         assertArrayEquals(new long[]{4, 1}, reshaped.getShape());
     }
 
+     @Test
+    public void canStackScalars() {
+        BooleanTensor x = BooleanTensor.scalar(true);
+        BooleanTensor y = BooleanTensor.scalar(true);
+
+        assertThat(BooleanTensor.create(true, true), valuesAndShapesMatch(BooleanTensor.stack(0, x, y)));
+    }
+
+    @Test
+    public void canStackVectors() {
+        BooleanTensor x = BooleanTensor.create(true, false);
+        BooleanTensor y = BooleanTensor.create(true, false);
+
+        assertEquals(BooleanTensor.create(true, false, true, false).reshape(2, 2), BooleanTensor.stack(0, x, y));
+        assertEquals(BooleanTensor.create(true, true, false, false).reshape(2, 2), BooleanTensor.stack(1, x, y));
+    }
+
+    @Test
+    public void canStackMatrices() {
+        BooleanTensor x = BooleanTensor.create(true, false).reshape(1, 2);
+        BooleanTensor y = BooleanTensor.create(true, false).reshape(1, 2);
+
+        assertThat(BooleanTensor.create(true, false, true, false).reshape(2, 1, 2), valuesAndShapesMatch(BooleanTensor.stack(0, x, y)));
+        assertThat(BooleanTensor.create(true, false, true, false).reshape(1, 2, 2), valuesAndShapesMatch(BooleanTensor.stack(1, x, y)));
+        assertThat(BooleanTensor.create(true, true, false, false).reshape(1, 2, 2), valuesAndShapesMatch(BooleanTensor.stack(2, x, y)));
+    }
+
+    @Test
+    public void canStackIfDimensionIsNegative() {
+        BooleanTensor x = BooleanTensor.create(true, false).reshape(1, 2);
+        BooleanTensor y = BooleanTensor.create(true, false).reshape(1, 2);
+
+        assertThat(BooleanTensor.create(true, false, true, false).reshape(2, 1, 2), valuesAndShapesMatch(BooleanTensor.stack(-3, x, y)));
+        assertThat(BooleanTensor.create(true, false, true, false).reshape(1, 2, 2), valuesAndShapesMatch(BooleanTensor.stack(-2, x, y)));
+        assertThat(BooleanTensor.create(true, true, false, false).reshape(1, 2, 2), valuesAndShapesMatch(BooleanTensor.stack(-1, x, y)));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void cannotStackIfPositiveDimensionIsOutOfBounds() {
+        BooleanTensor x = BooleanTensor.create(true, false).reshape(1, 2);
+        BooleanTensor y = BooleanTensor.create(true, false).reshape(1, 2);
+        BooleanTensor.stack(3, x, y);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void cannotStackIfNegativeDimensionIsOutOfBounds() {
+        BooleanTensor x = BooleanTensor.create(true, false).reshape(1, 2);
+        BooleanTensor y = BooleanTensor.create(true, false).reshape(1, 2);
```

Comment:
it would be worth making your various `x`s and `y`s static final fields of the class.